### PR TITLE
Factor out writing of Xplanet control file

### PR DESCRIPTION
**Pull request by dl1jbe**
...and cleanup the code

### CODE REVIEW

File: src/clusterinfo.c [hunk 1]
```diff
@@ -248,127 +244,112 @@ void loadbandmap(void) {
...
 
-		    if (*color != '\0') {
-			sprintf(marker_out, "%4d   %4d   \"%s\"   color=%s\n",
-				lat, lon, callcopy, color);
+		dx = dxcc_by_index(ctynr);
+		lon = (int)(dx -> lon) * -1;
+		lat = (int)(dx -> lat);
 
-			fputs(marker_out, fp);
+		if (spot_age[j] > 15)
```
Comment (zcsahok) [inline src/clusterinfo.c:279]: how about
```
char *color;
static char *bandcolor[NBANDS] = {"Red","Green","Blue",...., NULL /*for OOB*/};

color = NULL;  // probably not needed
if (age>N) { color = "Green"; }
else { color = bandcolor[freq2band(spot_freq_khz*1000)];
}
if (color != NULL) { fprintf(...); }
```

File: src/clusterinfo.c [hunk 2]
```diff
@@ -248,127 +244,112 @@ void loadbandmap(void) {
...
-
-			} else {
-			    strcat(color, "Cyan");
-			}
-		    }
+	    ctynr = getctynr(callcopy);		// CTY of station
+
+	    if (ctynr != 0 ) {
+		/* show no callsign if MARKERDOTS */
+		if (xplanet == 2)
```
Comment (zcsahok) [inline src/clusterinfo.c:272]: when refactoring logcfg parsing I thought it'd make sense to have constants for Xplanet modes, just didn't want to mess with it

**Review by zcsahok — COMMENTED** [review]
I tried Xplanet once and had a bad impression as it was showing the last (raw) spots that were randomly flickering, whereas the bandmap view showed a more stable set of spots. I changed the code to use bandmap for markers, but it was lost at some stage. Not sure if this PR affects this functionality.

**Comment by dl1jbe** [comment]
Both comments makes sense. As I had to fix the warning in clusterinfo,c in last days I just stumbled about the old xplanet code which was in really bad shape. So I tried as a first step to factor it out and clean it up to make clear what is going on. 
Furthermore the included call to getctynr() just used about 40% of TLFs CPU time (according to oprofile) - even if xplanet was disabled.

Further steps should follow:
- enumerated modes
- use the bandmap entries for it or at least
- search the spot_array backwards and stop as soon as we have picked up the 8 entries for export to Xplanet.

We should furthermore add a startup script for Xplanet with sensible settings to the distributed files.

At the moment I have a problem with the xplanet version in my distro as it ignores text output. So it shows only the dots. That needs to be fixed first here.

**Review by zcsahok — COMMENTED** [review]
Would it make sense to cache the results of `getctynr()`? Just thinking. The set of spotted calls is growing quite slowly (ok, maybe except during CQ WW) but they are evaluated repeatedly.

File: src/clusterinfo.c [hunk 3]
```diff
@@ -237,142 +232,95 @@ void loadbandmap(void) {
...
+		else {
+		    color = bandcolor[spot_band[j]];
+		}
+
+		if (color != NULL) {
+		    sprintf(marker_out, "%4d   %4d   \"%s\"   color=%s\n",
+			    lat, lon, callcopy, color);
 
-		    fclose(fp);
+		    fputs(marker_out, fp);
```
[the remark this answers is not in the document]
Comment (dl1jbe) [reply src/clusterinfo.c:294]: Right.

File: src/clusterinfo.c [hunk 4]
```diff
@@ -237,142 +232,95 @@ void loadbandmap(void) {
...
 
-	if ((fp = fopen(markerfile, "a")) == NULL) {
-	    TLF_LOG_INFO("Opening markerfile not possible.");
-	} else {
-	    if (strlen(xplanetmsg) > 20)
-		fputs(xplanetmsg, fp);
+	g_strlcat(xplanetmsg, "\"   color=Cyan\n", sizeof(xplanetmsg));
 
-	    fclose(fp);
+	if (strlen(xplanetmsg) > 20){
```
[the remark this answers is not in the document]
Comment (dl1jbe) [reply src/clusterinfo.c:315]: Good catch. Thanks. I look at these peace of code more than once and did not get it.
Fixed in next commit.

**Comment by dl1jbe** [comment]
> Would it make sense to cache the results of `getctynr()`? Just thinking. The set of spotted calls is growing quite slowly (ok, maybe except during CQ WW) but they are evaluated repeatedly.

I did some more checks with 'perf' to get better numbers for the overhead.
Running TLF for an hour on a typical afternoon cluster traffic I see the following numbers:

with MARKERS set:

* `find_best_mach` takes around 8..12% of the whole time TLF (or called functions) needs for running.
* The rest of the application code takes around the same percentage while TLF running idle.
* The final part is spent in different libraries (mostly ncurses related)

with MARKERS cleared

* The percentage of `find_best_match` drops to around 5 % . It is still called for every spot coming into the bandmap display.

TLF as whole spend around 1..2 % of the time of ONE core of my machine (4,3 GHz core i7).

So given the low overhead I think further work to spent a cache is not really needed.
Better would be to speed up the `find_best_match` function - maybe by sorting the prefixes alphabetically (or by length) which could speed up the search for all uses of that function.

**Pull request merged**